### PR TITLE
Port over all toolbox IDEs from the old repository, fixes #3

### LINKED ISSUE #3: Port over most supported products from the Gnome extension
See https://github.com/lunaryorn/jetbrains-search-provider

I guess the toolbox things would be good at least.

### PR DESCRIPTION
**Pull request by atomicptr**
I have only tested the ones I have installed myself but I figure its probably better to have them here already and fix them if they're broken (which might be somewhat unlikely given how easy the configuration is)

This solves #3

### CODE REVIEW

File: jetbrains-search-provider.py [hunk 1]
```diff
@@ -291,11 +291,51 @@ class ProviderDefinition:
...
+    ),
+    ProviderDefinition(
+        desktop_id="jetbrains-pycharm.desktop",
+        relative_obj_path="toolbox/pycharm",
+        config=ConfigLocation(vendor_dir="JetBrains", config_glob="PyCharm*"),
+    ),
+    ProviderDefinition(
+        desktop_id="jetbrains-rider.desktop",
+        relative_obj_path="toolbox/rider",
+        config=ConfigLocation(vendor_dir="JetBrains", config_glob="Rider*"),
```
Comment (lunaryorn) [inline jetbrains-search-provider.py:332]: I don't think Rider will work out of the box; as far as I remember it uses a different file to store recent projects.

File: jetbrains-search-provider.py [hunk 2]
```diff
@@ -291,11 +291,51 @@ class ProviderDefinition:
 # some distribution packages, etc.)
 PROVIDERS = [
     # Toolbox products
+    ProviderDefinition(
+        desktop_id="jetbrains-android-studio.desktop",
+        relative_obj_path="toolbox/android-studio",
+        config=ConfigLocation(vendor_dir="JetBrains", config_glob="AndroidStudio*"),
```
Comment (lunaryorn) [inline jetbrains-search-provider.py:297]: vendor_dir needs to be `Google` I think; please do check.

**Review by lunaryorn — CHANGES_REQUESTED** [review]

File: jetbrains-search-provider.py (hunk 1 above)
Comment (atomicptr) [reply jetbrains-search-provider.py]: I've added the ability to change the filename

File: jetbrains-search-provider.py (hunk 2 above)
Comment (atomicptr) [reply jetbrains-search-provider.py]: Whoops, thanks for the note!

**Comment by lunaryorn** [comment]
@atomicptr Thanks a bunch.

**Pull request merged**